src/models/team-application-access.ts

```diff
@@ -0,0 +1,239 @@
+	 *
+	 * @param {Number} teamId - Required: the team id.
+	 * @param {Object} [options={}] - extra pine options to use
+	 *
+	 * @fulfil {Object[]} - team application access
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.applicationAccess.getAllByTeam(1239948).then(function(teamApplicationAccesses) {
+	 * 	console.log(teamApplicationAccesses);
+	 * });
+	 */
+	const getAllByTeam = async function (
+		teamId: number,
+		options: BalenaSdk.PineOptions<BalenaSdk.TeamApplicationAccess> = {},
+	): Promise<BalenaSdk.TeamApplicationAccess[]> {
+		return sdkInstance.pine.get({
+			resource: 'team_application_access',
+			options: mergePineOptions(
+				{
+					$filter: {
+						team: teamId,
+					},
+				},
+				options,
+			),
+		});
+	};
+
+	/**
+	 * @summary Get team applications access
+	 * @name get
+	 * @public
+	 * @function
+	 * @memberof balena.models.team.applicationAccess
+	 *
+	 * @description This method get specific team application access.
+	 *
+	 * @param {Number} teamApplicationAccessId - Required: the team application access id.
+	 * @param {Object} [options={}] - extra pine options to use
+	 *
+	 * @fulfil {Object} - TeamApplicationAccess
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.applicationAccess.get(1239948).then(function(teamApplicationAccess) {
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 */
+	const get = async function (
+		teamApplicationAccessId: number,
+		options: BalenaSdk.PineOptions<BalenaSdk.TeamApplicationAccess> = {},
+	): Promise<BalenaSdk.TeamApplicationAccess | undefined> {
+		const teamApplicationAccess = await sdkInstance.pine.get({
+			resource: 'team_application_access',
+			id: teamApplicationAccessId,
+			options,
+		});
+		if (teamApplicationAccess == null) {
+			throw new Error('Team application access not found');
+		}
+		return teamApplicationAccess;
+	};
+
+	/**
+	 * @summary Add applications access to team
+	 * @name add
+	 * @public
+	 * @function
+	 * @memberof balena.models.team.applicationAccess.add
+	 *
+	 * @description This method add application access to team.
+	 *
+	 * @param {Number} teamId - Required: the team id the application access will be granted for.
+	 * @param {Number|String} applicationIdOrSlug - Required: application id or slug
+	 * @param {String} RoleName - Required: application membership role name (ApplicationMembershipRoles)
+	 *
+	 * @fulfil {Object} - TeamApplicationAccess
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.applicationAccess.add(1239948, 'MyAppSlug', 'developer').then(function(teamApplicationAccess) {
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 *
+	 * @example
+	 * balena.models.team.applicationAccess.add(1239948, 456789, 'observer').then(function(teamApplicationAccess) {
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 */
+	const add = async function (
+		teamId: number,
+		applicationIdOrSlug: number | string,
+		roleName: BalenaSdk.ApplicationMembershipRoles,
```

Comment:
I'm not entirely sure, but do you think it's worth updating this method to accept one of the following options?

For example:

```
applicationIdOrSlug: number | string | number[] | string[]  
roleName: BalenaSdk.ApplicationMembershipRoles | BalenaSdk.ApplicationMembershipRoles[]  
```

Or alternatively:

```
applicationSlugOrIdAndRole: { idOrSlug: number | string, roleName: BalenaSdk.ApplicationMembershipRoles }[]  
```
(just consider the idea not names that I gave as an example)

And then wrapping it in a batch function from utils. What do you think?